akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitAfterSpec.scala

```diff
@@ -115,6 +116,12 @@ class FlowSplitAfterSpec extends AkkaSpec {
       }
     }
 
+    "work with single elem splits" in assertAllStagesStopped {
+      Await.result(
+        Source(1 to 10).splitAfter(_ ⇒ true).lift.mapAsync(1)(_.runWith(Sink.head)).grouped(10).runWith(Sink.head),
```

Comment:
~~~scala
Source(1 to 10)
  .splitAfter(_ ⇒ true)
  .concatSubstreams
  .grouped(20)
  .runWith(Sink.head)
  .futureValue should ===(1 to 10)
~~~